pkg/operator/agent/agent.go

```diff
@@ -201,7 +201,11 @@ func (a *Agent) createAgentDaemonSet(namespace, agentImage string) error {
 		if !kserrors.IsAlreadyExists(err) {
 			return fmt.Errorf("failed to create rook-agent daemon set. %+v", err)
 		}
-		logger.Infof("rook-agent daemonset already exists")
+		logger.Infof("rook-agent daemonset already exists, updating ...")
+		_, err = a.clientset.Extensions().DaemonSets(namespace).Update(ds)
```

Comment:
what is the update behavior for the daemonSet, will it do a rolling/controlled upgrade across all the rook-agent pod instances?

Reply:
Kubernetes does a rolling update on changes on the rook-agent Pods, as seen here: https://github.com/galexrt/rook/blob/master/pkg/operator/agent/agent.go#L104-L106.